packages/client/tests/functional/find-unique-or-throw-batching/tests.ts

```diff
@@ -0,0 +1,43 @@
+import { faker } from '@faker-js/faker'
+
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+class NotFoundError extends Error {}
+declare let prisma: PrismaClient
+
+const id = faker.random.alphaNumeric(11)
+const missing = faker.random.alphaNumeric(11)
+
+testMatrix.setupTestSuite(
+  (suiteConfig, suiteMeta) => {
+    beforeEach(async () => {
+      await prisma.user.deleteMany()
+      await prisma.user.create({
+        data: {
+          id,
+        },
+      })
+    })
+
+    test('batched errors are serialized properly', async () => {
+      const found = prisma.user.findUniqueOrThrow({ where: { id } })
+      const foundToo = prisma.user.findUnique({ where: { id } })
+      const notFound = prisma.user.findUniqueOrThrow({ where: { id: missing } })
+      const result = await Promise.allSettled([found, foundToo, notFound])
+      expect(result).toEqual([
+        { status: 'fulfilled', value: { id: id } },
+        { status: 'fulfilled', value: { id: id } },
+        { reason: new NotFoundError('No User found'), status: 'rejected' },
```

Comment:
```suggestion
        { reason: exepct.objectMatching({ code: 'P2025' }) },
```

Reply:
Unfortunately, this doesn't work directly (even while replacing the typo), and I cannot find any other example at first glance, where that construct is used